app/controllers/rooms_controller.rb

```diff
@@ -0,0 +1,38 @@
+class RoomsController < ApplicationController
+    before_action :authenticate_user!
+  def create
+    @room = Room.create
+    @entry1 = Entry.create(:room_id => @room.id, :user_id => current_user.id)
+    @entry2 = Entry.create(params.require(:entry).permit(:user_id, :room_id).merge(:room_id => @room.id))
+    redirect_to "/rooms/#{@room.id}"
+  end
+
+  def show
+    @room = Room.find(params[:id])
+    if Entry.where(:user_id => current_user.id, :room_id => @room.id).present?
+      @messages = @room.messages
+      @message = Message.new
+      @entries = @room.entries
```

Comment:
変数が増えると混乱を招く可能性が高いので
viewのファイルで直接 `@room.messages` や `@room.entries` を参照したほうが良いかなと思います！
```suggestion
      @message = Message.new
```